File: packages/js/core/src/types/ResolveUriResult.ts
import { Api, Uri } from ".";
import { ResolveUriError, UriResolutionHistory } from "..";

export type ResolveUriResult = {
  api?: Api;
  uri?: Uri;
  uriHistory: UriResolutionHistory;
  error?: {
    type: ResolveUriError;
    error?: Error;
  };
};

File: packages/py/core/core/uri_resolution/core/types/resolve_uri_result.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Optional

from ....types import Uri, Wrapper
from . import ResolveUriError, UriResolutionHistory


@dataclass(slots=True, kw_only=True)
class ResolveUriResult:
    uri_history: UriResolutionHistory
    api: Optional[Wrapper] = None
    uri: Optional[Uri] = None
    error: Optional[ResolveUriError] = None
